### PR TITLE
dev-libs/libutf8proc: bump to version 2.2.0

### PR DESCRIPTION
**Pull request by ConiKost**
Closes: https://bugs.gentoo.org/667160
Signed-off-by: Conrad Kostecki <conrad@kostecki.com>
Package-Manager: Portage-2.3.49, Repoman-2.3.10

### CODE REVIEW

File: dev-libs/libutf8proc/libutf8proc-2.2.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+inherit cmake-utils toolchain-funcs
+
+DESCRIPTION="A C library that provides various operations for data in the UTF-8 encoding"
+HOMEPAGE="https://juliastrings.github.io/utf8proc/"
+SRC_URI="https://github.com/JuliaStrings/${MY_PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0/${PV}"
+KEYWORDS="~amd64 ~arm ~hppa ~ia64 ~ppc ~ppc64 ~sparc ~x86"
+IUSE="test"
```
Comment (Zlogene) [inline dev-libs/libutf8proc/libutf8proc-2.2.0.ebuild]: not being used

File: dev-libs/libutf8proc/libutf8proc-2.2.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+# Tests do need internet access
+RESTRICT="test"
+
+S="${WORKDIR}/${MY_PN}-${PV}"
+
+DOCS=(
```
Comment (Zlogene) [inline dev-libs/libutf8proc/libutf8proc-2.2.0.ebuild]: dodoc in src_install

File: dev-libs/libutf8proc/libutf8proc-2.2.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+	emake AR="$(tc-getAR)" CC="$(tc-getCC)"
+}
+
+src_test() {
+	emake check
+}
+
+src_install() {
+	emake LIBDIR_GENTOO="$(get_libdir)" PREFIX_GENTOO="${ED%/}" install
+	einstalldocs
```
Comment (Zlogene) [inline dev-libs/libutf8proc/libutf8proc-2.2.0.ebuild]: see above

File: dev-libs/libutf8proc/libutf8proc-2.2.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+DOCS=(
+	"lump.md"
+	"NEWS.md"
+	"README.md"
+)
+
+PATCHES=( "${FILESDIR}/${P}-fix-path.patch" )
+
+src_prepare() {
+	cmake-utils_src_prepare
```
Comment (Zlogene) [inline dev-libs/libutf8proc/libutf8proc-2.2.0.ebuild]: call `tc-export AR CC` then I believe src_compile() can be removed

File: dev-libs/libutf8proc/libutf8proc-2.2.0.ebuild (hunk 1 above)
Comment (ConiKost) [reply dev-libs/libutf8proc/libutf8proc-2.2.0.ebuild]: So for enabling tests, I don't need `IUSE="test"`?

File: dev-libs/libutf8proc/libutf8proc-2.2.0.ebuild (hunk 2 above)
Comment (ConiKost) [reply dev-libs/libutf8proc/libutf8proc-2.2.0.ebuild]: Ok, will do. Could you explain me, why? I don't see any difference.

File: dev-libs/libutf8proc/libutf8proc-2.2.0.ebuild (hunk 3 above)
Comment (ConiKost) [reply dev-libs/libutf8proc/libutf8proc-2.2.0.ebuild]: Done.

File: dev-libs/libutf8proc/libutf8proc-2.2.0.ebuild (hunk 4 above)
Comment (ConiKost) [reply dev-libs/libutf8proc/libutf8proc-2.2.0.ebuild]: Yes, this works. Thanks!

File: dev-libs/libutf8proc/libutf8proc-2.2.0.ebuild (hunk 1 above)
Comment (Zlogene) [reply dev-libs/libutf8proc/libutf8proc-2.2.0.ebuild]: @ConiKost `USE=test` is for deps that are needed ro run the test suit

File: dev-libs/libutf8proc/libutf8proc-2.2.0.ebuild (hunk 2 above)
Comment (Zlogene) [reply dev-libs/libutf8proc/libutf8proc-2.2.0.ebuild]: what is the point to define `DOCS` and then call its installation, `DOCS` is good once you are using default installation implementation (undeclared one), for the rest, it is pointless

File: dev-libs/libutf8proc/libutf8proc-2.2.0.ebuild (hunk 1 above)
Comment (ConiKost) [reply dev-libs/libutf8proc/libutf8proc-2.2.0.ebuild]: Ah, thanks for clarification. I see, what you mean.

File: dev-libs/libutf8proc/libutf8proc-2.2.0.ebuild (hunk 2 above)
Comment (ConiKost) [reply dev-libs/libutf8proc/libutf8proc-2.2.0.ebuild]: You could ask the same question in reverse ;-) Both achieve exactly the same, so IMHO it doesn't matter at all. I think, you could see faster the definied DOCS than somewhere in src_*. I would agree, when one would define DOCS and also some `dodoc`. Anyway, will have a eye on that.

**Comment by ConiKost** [comment]
This got apprently rev bumped, so this PR is now useless. Closing.

**Pull request closed without merging**